File: app/src/main/java/com/footinit/baseavd/AvdApp.java
package com.footinit.baseavd;

import android.app.Application;

import com.footinit.baseavd.di.component.ApplicationComponent;
import com.footinit.baseavd.di.component.DaggerApplicationComponent;
import com.footinit.baseavd.di.module.ApplicationModule;

import javax.inject.Inject;
import uk.co.chrisjenx.calligraphy.CalligraphyConfig;

/**
 * Created by Abhijit on 28-11-2017.
 */

public class AvdApp extends Application {

    @Inject
    CalligraphyConfig calligraphyConfig;

    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        //Instantiate ApplicationComponent
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();

        CalligraphyConfig.initDefault(calligraphyConfig);
    }

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: app/src/main/java/com/footinit/baseavd/AvdApp.kt
package com.footinit.baseavd

import android.app.Application
import com.footinit.baseavd.di.appModule
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

class AvdApp: Application() {
    override fun onCreate() {
        super.onCreate()

        startKoin {
            androidContext(this@AvdApp)
            androidLogger()
            modules(appModule)
        }
    }
}